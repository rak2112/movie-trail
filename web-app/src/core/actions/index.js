export * from './api';
export * from './entities';
export * from './movie-details';
export * from './user';