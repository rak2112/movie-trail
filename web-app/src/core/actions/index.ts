export * from './api';
export * from './entities';
export * from './movie-details';
export * from './search-movies';
export * from './user';