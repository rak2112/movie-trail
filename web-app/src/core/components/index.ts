export * from './carousel';
export * from './form-input';
export * from './form-validation';
export * from './imageLoader';
export * from './load-error';
export * from './loader';
export * from './loader-wrapper';
export * from './movie';
export * from './movie-list';
export * from './pagination';
export * from './toast';
export * from './user-actions';

