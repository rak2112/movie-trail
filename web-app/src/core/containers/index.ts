export * from './error';
export * from './user-actions';