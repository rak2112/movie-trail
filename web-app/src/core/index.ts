// // components
// export * from './components/carousel';
// export * from './components/form-input';
// export * from './components/form-validation';
// export * from './components/imageLoader';
// export * from './components/load-error';
// export * from './components/loader';
// export * from './components/loader-wrapper';
// export * from './components/movie';
// export * from './components/movie-list';
// export * from './components/pagination';
// export * from './components/toast';
// export * from './components/user-actions';

// // containers
// export {default as Error } from './containers/error';

// // utils
// export * from './utils/movies-service';
// export * from './utils/util-service';
// export * from './utils/user-service';
// export * from './utils/localStorage';
// export * from './utils/http';

// // store 
// export * from './store/configureStore';

// // // selectors
// export * from './selectors/movie-details';
// export * from './selectors/movies';


// // interfaces 
// export * from './interfaces';

// // routes
// export * from './routes';

// // reducders
// export * from './reducers';

// // actions
// export * from './actions/api';
// export * from './actions/entities';
// export * from './actions/movie-details';
// export * from './actions/user';