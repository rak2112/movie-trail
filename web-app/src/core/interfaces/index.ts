type MovieType =
  | 'Action'
  | 'Adventure'
  | 'Animation'
  | 'Comedy'
  | 'Crime'
  | 'Documentry'
  | 'Drama'
  | 'Family'
  | 'Fantasy'
  | 'History'
  | 'Horror'
  | 'Music'
  | 'Mystery'
  | 'Romance'
  | 'Science Fiction'
  | 'TV Movie'
  | 'Thriller'
  | 'War'
  | 'Western';

export type UserMovieType= 'FAVORITE' | 'WATCHLIST';

export type UserView = 'Favorites' | 'Watchlist';

export interface Api {
  isFetching?: boolean;
  hasError?: boolean;
  errorMessage?: string;
}; 

export interface FormValues {
  name?: string;
  email: string;
  password: string;
  confirmPassword?: string;
  token?: string;
}

export interface Language {
  id: string;
  name: string;
}

export interface Movie {
  id: number;
  backdrop_path: string;
  genre_ids: number[];
  genres: Genre[];
  homepage: string;
  overview: string;
  poster_path?: string;
  runtime: string;
  release_date: string;
  spoken_languages: Language[];
  status: string;
  title: string;
  type?: string;
  vote_average: string;
};

export interface Movies {
  pageNo: number;
  totalPages: number;
  totalResults: number;
  results: Movie[];
  redirection? : boolean;
};

export interface MoviesData {
  pageNo: number;
  total_results: number;
  total_pages: number;
  results?: Movie[];
};

export interface MoviesRes {
  movies: Movie[]
};

export interface Genre {
  id: number;
  name: MovieType;
};

export interface GenresMap {
  [id: number]: Genre;
};

export interface UserMovie extends Movie {
  _id?: string;
  movieType?: UserMovieType;
}

export interface User {
  email?: string;
  displayName?: string;
  loggedIn?: boolean;
  movies?: UserMovie[];
  resetMessage?: string;
  token?: string;
}

export interface UserMovies {
  [id: string]: UserMovie;
}
export interface UserMovieMap {
  favorites: UserMovies;
  watchlist: UserMovies;
}

export interface Cast {
  character: string;
  id: string;
  name: string;
  profile_path?: string;
}

export interface Crew extends Cast {
  department: string;
  job: string;
}

export interface Poster {
  file_path: string;
}

export interface Persons {
  id?: string;
  cast: Cast[];
  crew: Crew[];
}

export interface Posters {
  id?: string;
  backdrops: Poster[];
  posters: Poster[];
}

export interface Video {
  id: string;
  key: string;
  name: string;
}

export interface Videos {
  id: string;
  results: Video[]
}

export interface MovieDetail {
  isFetching?: boolean;
  hasError?: boolean;
  details: Movie;
  images: Posters;
  persons: Persons;
  videos: Videos
}