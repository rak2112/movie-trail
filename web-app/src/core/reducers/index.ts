import { routerReducer } from 'react-router-redux';
import { combineReducers } from 'redux';
import { reducer as formReducer } from 'redux-form'

import { api, ApiState } from './api';
import { genres, GenresState } from './genres';
import { movieDetails, MovieDetailState } from './movie-details';
import { movies, MoviesState } from './movies';
import { user, UserState } from './user';


export interface State {
  readonly api: ApiState;
  readonly genres: GenresState
  readonly movieDetails: MovieDetailState;
  readonly movies: MoviesState;
  readonly user: UserState;
}

const rootReducer = combineReducers({
  api,
  form: formReducer,
  genres,
  movieDetails,
  movies,
  routing: routerReducer,
  user
});

export default rootReducer;