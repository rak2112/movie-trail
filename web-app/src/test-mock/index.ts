import { Genre, Movie, Movies, UserMovieMap } from '../core/interfaces';

import { State } from '../core/reducers';

export const defaultMoviesState = {
  addMovie: jest.fn(),
  deleteMovie: jest.fn(),
  genres:[
    {id:2, name: "Action"} as Genre
  ], 
  loadMovies: jest.fn(),
  movies: {
    totalPages: 1, pageNo: 1, totalResults: 3, 
    results: [{
      id: 2,
      backdrop_path: 'path',
      genre_ids: [1, 2],
      genres: [],
      homepage: 'homepage',
      overview: 'overview',
      runtime: '107',
      release_date: '2-04-19',
      spoken_languages: [],
      status: 'status',
      title: 'title',
      vote_average: 'vote_avg'
    },
    {
      id: 3,
      backdrop_path: 'path',
      genre_ids: [1, 2],
      genres: [],
      homepage: 'homepage',
      overview: 'overview',
      runtime: '107',
      release_date: '2-04-19',
      spoken_languages: [],
      status: 'status',
      title: 'title',
      vote_average: 'vote_avg'
    }]
  } as Movies,
  userMovies: {
    favorites: {
      2: {
        id: 2,
        _id: '23',
        backdrop_path: 'path',
        genre_ids: [1, 2],
        genres: [],
        homepage: 'homepage',
        movieType: 'FAVORITE',
        overview: 'overview',
        runtime: '107',
        release_date: '2-04-19',
        spoken_languages: [],
        status: 'status',
        title: 'title',
        vote_average: 'vote_avg'
      }
    },
    watchlist: {}
  } as UserMovieMap
};

export const defaultState = {
  api: { isFetching: true, hasError: false },
  genres: [{id:2, name: 'Action'} as Genre ],
  movies: null,
  movieDetails: null,
  searchedMovies: null,
  user: null,
  userMovies: {}
} as State;

export const defaultMovies = {
  totalPages: 1, pageNo: 1, totalResults: 3, 
  results: [{
    id: 2,
    backdrop_path: 'path',
    genre_ids: [1, 2],
    genres: [],
    homepage: 'homepage',
    overview: 'overview',
    runtime: '107',
    release_date: '2-04-19',
    spoken_languages: [],
    status: 'status',
    title: 'title',
    vote_average: 'vote_avg'
  },
  {
    id: 3,
    backdrop_path: 'path',
    genre_ids: [1, 2],
    genres: [],
    homepage: 'homepage',
    overview: 'overview',
    runtime: '107',
    release_date: '2-04-19',
    spoken_languages: [],
    status: 'status',
    title: 'title',
    vote_average: 'vote_avg'
  }] as Movie[]
} as Movies;