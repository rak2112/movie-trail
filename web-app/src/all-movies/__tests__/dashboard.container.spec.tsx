import React, { ComponentClass } from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter, Route } from 'react-router';
import { create } from 'react-test-renderer';
import { Store } from 'redux';
import configureMockStore from 'redux-mock-store';

import { State } from 'src/core/reducers';
import { defaultMovies, defaultState } from '../../test-mock';
import ConnectedMoviesContainer from '../dashboard.container';


describe(`Movies Container`, () => {
  let initialState: State;
  let store: Store;

  beforeEach(() => {
    initialState = defaultState;
  });

  it(`if the state is in loading`, () => {
    store = configureMockStore()(initialState);

    const component = create(
      <Provider store={store}>
        <MemoryRouter initialIndex={0} initialEntries={['/movies']}>
          <Route path={'/movies'} component={ConnectedMoviesContainer} />
        </MemoryRouter>
      </Provider>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it(`if the state is loaded and has data`, () => {
    initialState = {
      ...defaultState, 
      api: {...defaultState.api, isFetching: false},
      movies: defaultMovies
    };

    store = configureMockStore()(initialState);

    const component = create(
      <Provider store={store}>
        <MemoryRouter initialIndex={0} initialEntries={['/movies']}>
          <Route path={'/movies'} component={ConnectedMoviesContainer} />
        </MemoryRouter>
      </Provider>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

});