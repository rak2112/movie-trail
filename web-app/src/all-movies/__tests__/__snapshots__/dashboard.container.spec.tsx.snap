// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies Container if the state is in loading 1`] = `
<div
  className="alert alert-info"
  role="alert"
>
  <i
    className="glyphicon glyphicon-repeat gly-spin"
  />
  <span>
    This may take a moment, please wait.
  </span>
</div>
`;

exports[`Movies Container if the state is loaded and has data 1`] = `
Array [
  <input
    className="form-control"
    onChange={[Function]}
    onMouseLeave={[Function]}
    placeholder="Search Movies"
    style={
        Object {
            "background": "#fff",
            "borderRaidus": 0,
            "height": "45px",
          }
    }
    type="text"
    value=""
/>,
  <div
    className="css-5zmk01"
>
    <div
        className="css-182pldt"
    >
        <div
            className="Toastify"
        />
    </div>
    <div
        className="css-h5tpbd"
    >
        <div
            className="css-bxd89m"
        >
            <a
                href="/movie-details/2"
                onClick={[Function]}
            >
                <img
                    alt="main image"
                    src="https://image.tmdb.org/t/p/w500path "
                />
            </a>
            <div
                className="css-1nkcf6k"
            >
                <div
                    className="user-actions"
                >
                    <span
                        className="rating"
                    >
                        vote_avg
                        <span
                            className="glyphicon glyphicon-star"
                        />
                    </span>
                    <span
                        className="year"
                    >
                        <span
                            className="glyphicon glyphicon-calendar"
                        />
                        2019
                    </span>
                </div>
            </div>
            <div
                className="css-1w2z22x"
            >
                <h3>
                    title
                </h3>
                <span
                    className="genre"
                >
                    Action
                </span>
                <p>
                    <a
                        href="/movie-details/2"
                        onClick={[Function]}
                    >
                        View Details
                    </a>
                </p>
            </div>
        </div>
    </div>
    <div
        className="css-h5tpbd"
    >
        <div
            className="css-bxd89m"
        >
            <a
                href="/movie-details/3"
                onClick={[Function]}
            >
                <img
                    alt="main image"
                    src="https://image.tmdb.org/t/p/w500path "
                />
            </a>
            <div
                className="css-1nkcf6k"
            >
                <div
                    className="user-actions"
                >
                    <span
                        className="rating"
                    >
                        vote_avg
                        <span
                            className="glyphicon glyphicon-star"
                        />
                    </span>
                    <span
                        className="year"
                    >
                        <span
                            className="glyphicon glyphicon-calendar"
                        />
                        2019
                    </span>
                </div>
            </div>
            <div
                className="css-1w2z22x"
            >
                <h3>
                    title
                </h3>
                <span
                    className="genre"
                >
                    Action
                </span>
                <p>
                    <a
                        href="/movie-details/3"
                        onClick={[Function]}
                    >
                        View Details
                    </a>
                </p>
            </div>
        </div>
    </div>
    <footer
        className="css-dpqy78"
    >
        <ul
            className="pagination"
        >
            <li
                className="disabled"
                onClick={[Function]}
            >
                <a
                    className=""
                    href="#"
                >
                    «
                </a>
            </li>
            <li
                className="disabled"
                onClick={[Function]}
            >
                <a
                    className=""
                    href="#"
                >
                    ⟨
                </a>
            </li>
            <li
                className="active"
                onClick={[Function]}
            >
                <a
                    className="undefined"
                    href="#"
                >
                    1
                </a>
            </li>
            <li
                className="disabled"
                onClick={[Function]}
            >
                <a
                    className=""
                    href="#"
                >
                    ⟩
                </a>
            </li>
            <li
                className="disabled"
                onClick={[Function]}
            >
                <a
                    className=""
                    href="#"
                >
                    »
                </a>
            </li>
        </ul>
    </footer>
</div>,
]
`;
